### PR TITLE
integration-tests: Add actor testing framework, ACD actor tests

### PR DESCRIPTION
**Pull request by mslipper**
Metadata:

- Fixes ENG-1662
- Fixes ENG-1664
- Fixes ENG-1663

Adds the actor testing framework and implements the actor tests for ACD. The tests are:

- Deposits
- Transfers
- Uniswap
- NFTs
- The DoS Contract

For information on how the framework works, see [integration-tests/actor-tests/README.md](integration-tests/actor-tests/README.md).

### CODE REVIEW

File: integration-tests/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "lint:check": "eslint .",
     "build": "hardhat compile",
     "test:integration": "hardhat --network optimism test",
+    "test:actor": "IS_LIVE_NETWORK=true ts-node actor-tests/lib/runner.ts",
```
Comment (tynes) [inline integration-tests/package.json:14]: I don't see `IS_LIVE_NETWORK` consumed anywhere in the files added to this PR

File: integration-tests/actor-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+# HELP actor_successful_actor_runs_total Count of total successful actor runs.
+# TYPE actor_successful_actor_runs_total counter
+actor_successful_actor_runs_total{actor_name="value_sender"} 40
+
+# HELP actor_failed_actor_runs_total Count of total failed actor runs.
+# TYPE actor_failed_actor_runs_total counter
+```
+
+## CLI
```
Comment (tynes) [inline integration-tests/actor-tests/README.md:151]: I think this should be at the top of the `README`

File: integration-tests/actor-tests/lib/actor.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    }
+    this.logger = new WorkerLogger(this.actor.name, workerId)
+  }
+
+  bench = async (name: string, cb: () => Promise<void>) => {
+    const metricLabels = {
+      actor_name: sanitizeForMetrics(this.actor.name),
+      bench_name: sanitizeForMetrics(name),
+    }
+    const start = performance.now()
```
Comment (tynes) [inline integration-tests/actor-tests/lib/actor.ts:86]: Getting an error when trying to run the tests saying that `performance` is not defined

Comment (mslipper) [reply integration-tests/actor-tests/lib/actor.ts:86]: Ah, looks like `performance` is a global in Node 16+, but requires a package import before. I'll fix this.

File: integration-tests/package.json (hunk 1 above)
Comment (mslipper) [reply integration-tests/package.json:14]: It's used by the `OptimismEnv` class.

**Comment by tynes** [comment]
This PR generally looks good to me, there are a few linting warnings in it

**Pull request merged**